### PR TITLE
[improvement](meta) Switch meta serialization to gson 4

### PR DESCRIPTION
**Pull request by iszhangpch**
## Proposed changes

Issue Number:

Switch meta serialization to gson
Contains the following classes：

RoutineLoadJob
UserPropertyInfo
AnalyzeDeletionLog
CreateTableInfo
HbPackage
~~PartitionPersistInfo~~
ReplicaPersistInfo
RoutineLoadOperation

### CODE REVIEW

**Comment by iszhangpch** [comment]
run buildall

**Comment by iszhangpch** [comment]
run buildall

**Comment by iszhangpch** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/load/routineload/RoutineLoadProgress.java [hunk 1]
```diff
@@ -44,25 +47,29 @@ public RoutineLoadProgress(LoadDataSourceType loadDataSourceType) {
...
     public static RoutineLoadProgress read(DataInput in) throws IOException {
-        RoutineLoadProgress progress = null;
-        LoadDataSourceType type = LoadDataSourceType.valueOf(Text.readString(in));
-        if (type == LoadDataSourceType.KAFKA) {
-            progress = new KafkaProgress();
+        if (Env.getCurrentEnvJournalVersion() < FeMetaVersion.VERSION_137) {
+            RoutineLoadProgress progress = null;
+            LoadDataSourceType type = LoadDataSourceType.valueOf(Text.readString(in));
+            if (type == LoadDataSourceType.KAFKA) {
+                progress = new KafkaProgress();
```
Comment (dataroaring) [inline fe/fe-core/src/main/java/org/apache/doris/load/routineload/RoutineLoadProgress.java:54]: We should handle like this in gsonPostProcess.

File: fe/fe-core/src/main/java/org/apache/doris/load/routineload/RoutineLoadProgress.java [hunk 2]
```diff
@@ -44,25 +47,29 @@ public RoutineLoadProgress(LoadDataSourceType loadDataSourceType) {
...
-        progress.readFields(in);
-        return progress;
     }
 
     @Override
     public void write(DataOutput out) throws IOException {
-        // ATTN: must write type first
-        Text.writeString(out, loadDataSourceType.name());
+        Text.writeString(out, GsonUtils.GSON.toJson(this));
     }
```
Comment (dataroaring) [inline fe/fe-core/src/main/java/org/apache/doris/load/routineload/RoutineLoadProgress.java:70]: Just remove write method.

File: fe/fe-core/src/main/java/org/apache/doris/load/routineload/RoutineLoadJob.java [hunk 3]
```diff
@@ -1799,68 +1833,29 @@ public boolean isFinal() {
...
     public static RoutineLoadJob read(DataInput in) throws IOException {
-        RoutineLoadJob job = null;
-        LoadDataSourceType type = LoadDataSourceType.valueOf(Text.readString(in));
-        if (type == LoadDataSourceType.KAFKA) {
-            job = new KafkaRoutineLoadJob();
+        if (Env.getCurrentEnvJournalVersion() < FeMetaVersion.VERSION_137) {
+            RoutineLoadJob job = null;
+            LoadDataSourceType type = LoadDataSourceType.valueOf(Text.readString(in));
+            if (type == LoadDataSourceType.KAFKA) {
+                job = new KafkaRoutineLoadJob();
```
Comment (dataroaring) [inline fe/fe-core/src/main/java/org/apache/doris/load/routineload/RoutineLoadJob.java:1840]: handle like this in gsonPostProcess.

File: fe/fe-core/src/main/java/org/apache/doris/persist/ReplicaPersistInfo.java [hunk 4]
```diff
@@ -81,29 +87,46 @@ public static ReplicaOperationType findByValue(int value) {
     }
 
     // required
+    @SerializedName("op")
     private ReplicaOperationType opType;
```
Comment (dataroaring) [inline fe/fe-core/src/main/java/org/apache/doris/persist/ReplicaPersistInfo.java:91]: please handle field of ReplicaOperationType.

File: fe/fe-core/src/main/java/org/apache/doris/persist/CreateTableInfo.java [hunk 5]
```diff
@@ -58,17 +60,22 @@ public Table getTable() {
...
+            return createTableInfo;
+        } else {
+            return GsonUtils.GSON.fromJson(Text.readString(in), CreateTableInfo.class);
+        }
     }
 
+    @Deprecated
     private void readFields(DataInput in) throws IOException {
         dbName = ClusterNamespace.getNameFromFullName(Text.readString(in));
         table = Table.read(in);
```
Comment (dataroaring) [inline fe/fe-core/src/main/java/org/apache/doris/persist/CreateTableInfo.java:81]: handle this in gson post proessor.

**Comment by dataroaring** [comment]
run buildall

**Comment by iszhangpch** [comment]
run buildall

**Comment by iszhangpch** [comment]
run buildall

**Review by dataroaring — DISMISSED** [review]
LGTM

**Comment by iszhangpch** [comment]
run buildall

**Comment by iszhangpch** [comment]
run buildall

**Review by dataroaring — DISMISSED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/task/LoadTaskInfo.java [hunk 6]
```diff
@@ -130,7 +131,9 @@ default int getStreamPerNode() {
     }
 
     class ImportColumnDescs {
+        @SerializedName("des")
         public List<ImportColumnDesc> descs = Lists.newArrayList();
```
Comment (w41ter) [inline fe/fe-core/src/main/java/org/apache/doris/task/LoadTaskInfo.java:135]: `ImportColumnDesc` is not annotated with `SerializedName`.

File: fe/fe-core/src/main/java/org/apache/doris/mysql/privilege/UserPropertyInfo.java [hunk 7]
```diff
@@ -52,21 +58,23 @@ public List<Pair<String, String>> getProperties() {
...
-        UserPropertyInfo info = new UserPropertyInfo();
-        info.readFields(in);
-        return info;
+        if (Env.getCurrentEnvJournalVersion() < FeMetaVersion.VERSION_137) {
+            UserPropertyInfo info = new UserPropertyInfo();
+            info.readFields(in);
+            return info;
+        } else {
+            UserPropertyInfo uProp = GsonUtils.GSON.fromJson(Text.readString(in), UserPropertyInfo.class);
+            uProp.user = ClusterNamespace.getNameFromFullName(uProp.user);
```
Comment (w41ter) [inline fe/fe-core/src/main/java/org/apache/doris/mysql/privilege/UserPropertyInfo.java:67]: handle `user` in gson post processor.

File: fe/fe-core/src/main/java/org/apache/doris/persist/CreateTableInfo.java [hunk 8]
```diff
@@ -58,17 +60,22 @@ public Table getTable() {
...
     public static CreateTableInfo read(DataInput in) throws IOException {
-        CreateTableInfo createTableInfo = new CreateTableInfo();
-        createTableInfo.readFields(in);
-        return createTableInfo;
+        if (Env.getCurrentEnvJournalVersion() < FeMetaVersion.VERSION_137) {
+            CreateTableInfo createTableInfo = new CreateTableInfo();
+            createTableInfo.readFields(in);
+            return createTableInfo;
+        } else {
+            return GsonUtils.GSON.fromJson(Text.readString(in), CreateTableInfo.class);
```
Comment (w41ter) [inline fe/fe-core/src/main/java/org/apache/doris/persist/CreateTableInfo.java:74]: handle `dbName` in gson post processor.

**Review by w41ter — CHANGES_REQUESTED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/load/routineload/RoutineLoadJob.java [hunk 9]
```diff
@@ -172,23 +180,29 @@ public boolean isFinalState() {
     protected Expr whereExpr; // optional
     protected Separator columnSeparator; // optional
     protected Separator lineDelimiter;
+    @SerializedName("dtcn")
     protected int desireTaskConcurrentNum; // optional
+    @SerializedName("st")
     protected JobState state = JobState.NEED_SCHEDULE;
     @Getter
+    @SerializedName("dst")
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/load/routineload/RoutineLoadJob.java:188]: naming `dsrc`
dst usually means destination

File: fe/fe-core/src/main/java/org/apache/doris/persist/CreateTableInfo.java (hunk 8 above)
Comment (iszhangpch) [reply fe/fe-core/src/main/java/org/apache/doris/persist/CreateTableInfo.java:74]: already done.

**Comment by iszhangpch** [comment]
run buildall

**Comment by iszhangpch** [comment]
run buildall

**Comment by iszhangpch** [comment]
run buildall

**Comment by iszhangpch** [comment]
run buildall

**Comment by iszhangpch** [comment]
run buildall

**Comment by iszhangpch** [comment]
run buildall

**Comment by iszhangpch** [comment]
run buildall

**Comment by iszhangpch** [comment]
run buildall

**Review by dataroaring — DISMISSED** [review]
LGTM

**Review by w41ter — DISMISSED** [review]

**Comment by dataroaring** [comment]
run buildall

**Review by w41ter — APPROVED** [review]

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**